components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/EventPublisher.java

```diff
@@ -145,6 +147,33 @@ public void sendEvent(Event event) {
         if (isPolled || !EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop()) {
             process(event);
         }
+
+        if(EventPublisherServiceValueHolder.getEventManagementService().getManagementModeInfo().getMode() == Mode.HA){
+            if(!isPolled && EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop()){
```

Comment:
Seems you could take out, if(!isPolled) and then check for the EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop(). That would make the code clear AFAIS.